game/source_engine_nav.ksy

```diff
@@ -0,0 +1,468 @@
+  - id: place_name_array
+    type: place_name
+    repeat: expr
+    repeat-expr: place_count
+    if: 'version >= 5'
+    doc: Container for $place_count place names. Added in NAV version 5.
+  # Unnamed areas boolean.
+  - id: has_unnamed_areas
+    type: u1 # The value is interpreted as a boolean in the Source-Engine C++ code, but it is actually stored as a byte.
+    if: 'version >= 12'
+    doc: Determines if NAV file has unnamed areas. Added in NAV version 12.
+  # Area data.
+  - id: area_count
+    type: u4
+    doc: Amount of areas stored in the nav mesh.
+  - id: area_array
+    type: nav_area
+    repeat: expr
+    repeat-expr: area_count
+    doc: Container for stored navigation areas.
+  # Ladders
+  - id: ladder_count
+    type: u4
+    if: '_root.version >= 6'
+    doc: Amount of ladders stored in the nav mesh. Added in NAV version 6.
+  - id: ladder_array
+    type: nav_ladder
+    repeat: expr
+    repeat-expr: ladder_count
+    if: '_root.version >= 6'
+    doc: Container for ladder areas stored in the nav mesh. Added in NAV version 6.
+  - id: custom_data
+    size-eos: true
+    if: '_root.version >= 10 and _root.subversion != 0'
+    doc: Custom data appended by a specialized implementation of the NAV format. Should only be parsed if NAV file has a subversion.
+types:
+  place_name:
+    doc: |
+      A label of a location in a map. They are defined by the map.
+
+      Place names are stored as Length-String value pairs.
+    seq:
+      - id: length
+        type: u2
+        doc: Length of the place name.
+      - id: name
+        type: str
+        size: length
+        encoding: ASCII
+        doc: The place name.
+  # Connection Pair
+  area_id_array:
+    seq:
+      - id: count
+        type: u4
+      - id: area_ids
+        type: u4
+        repeat: expr
+        repeat-expr: count
+  # ApproachInfo
+  approach_info:
+    doc: |
+      Datum point that determines the type of movement between areas. Removed in NAV version 15.
+
+      It is only used by the Counter-Strike video game series.
+
+    # The elements
+    seq:
+      - id: target_area_id
+        type: u4
+        doc: Area ID of area to approach to.
+      - id: approach_prev
+        type: u4
+        doc: Area ID of the area before the current approach area.
+      - id: type
+        type: u1
+        doc: Type of approach method.
+      - id: next
+        type: u4
+        doc: Area ID of area to approach to next.
+      - id: method
+        type: u1
+  # NavHidingSpot
+  hiding_spot:
+    seq:
+      - id: area_id
+        type: u4
+        if: '_root.version >= 2'
+        doc: Area ID of the hiding spot's parent area. Added in NAV version 2.
+      - id: hide_spot_position
+        type: f4
+        repeat: expr
+        repeat-expr: 3
+        if: '_root.version >= 1'
+        doc: The position of the Hiding spot (as a vector). Added in NAV version 1.
+      - id: attribute_flag
+        type: u1
```

Comment:
refactor `hide_spot_attribute` into a type of 8 fields of type `b1`.

Reply:
I converted the hide spot attribute flag into a 8-bit field.

However the area-bind attribute flag is written, read, and compared to as a number, and not as a bit flag, by the Source Engine, so I left that value as an unsigned byte type.